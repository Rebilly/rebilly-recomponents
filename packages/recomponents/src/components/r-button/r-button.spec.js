import {mount, shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RButton from './r-button.vue';

describe('r-button.vue', () => {
    it('renders props.msg when passed', () => {
        const msg = 'Apply';
        const wrapper = shallowMount(RButton, {
            propsData: {msg},
            mocks: {
                $t: () => msg,
            },
        });
        expect(wrapper.text()).toMatch(msg);
    });

    it('should render via SSR and match snapshot', async () => {
        const wrapper = renderToString(RButton, {
            slots: {
                default: '<strong>default button</strong>',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with type prop', () => {
        const wrapper = mount(RButton, {
            propsData: {
                type: 'danger',
            },
            slots: {
                badge: '<strong>danger button</strong>',
            },
        });

        expect(wrapper.classes()).toContain('r-button--type-warning');
    });

    it('should render component with size, fluid or disabled prop', () => {
        const $t = () => {
        };

        const wrapper = shallowMount(RButton, {
            propsData: {
                size: 'small',
                disabled: true,
                fluid: true,
            },
            slots: {
                badge: '<strong>small button</strong>',
            },
            mocks: {$t},
        });

        expect(wrapper.classes()).toContain('r-button--size-small');
        expect(wrapper.classes()).toContain('r-button--fluid');
        expect(wrapper.attributes().disabled).not.toBe(undefined);
    });
});
