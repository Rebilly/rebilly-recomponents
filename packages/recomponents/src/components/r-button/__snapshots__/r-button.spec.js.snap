// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-button.vue should render via SSR and match snapshot 1`] = `
<button title="" msg="Apply" class="r-component r-button r-button--size-regular r-button--type-default">
  <!----> Apply</button>
`;
